### PR TITLE
afc_timing: update driver interface and add new records.

### PR DESCRIPTION
**Pull request by ericonr**
Depends on https://github.com/lnls-dig/uhal/pull/57

### CODE REVIEW

File: utcaApp/Db/pos_calc.template [hunk 1]
```diff
@@ -64,6 +64,7 @@ record(ao,"$(P)$(R)PosKsum-SP"){
     field(OUT,"@asyn($(PORT))KSUM")
     field(ASG, "Interlock")
     field(PREC, 8)
+    field(HIHI, 1)
```
Comment (henriquesimoes) [inline utcaApp/Db/pos_calc.template]: This change does not seem related to the driver interface nor its new records. IMHO, this deserves its own commit, explaining the rationale for introducing such alarm now.

File: utcaApp/src/afc_timing.cpp [hunk 2]
```diff
@@ -33,50 +24,56 @@ class AFCTiming: public UDriver {
...
+          }, &ctl),
       dec(bars),
       ctl(bars)
     {
         auto v = find_device(port_number);
         dec.set_devinfo(v);
         ctl.set_devinfo(v);
 
+        parameter_props.at(p_ch_src).write_decoder_controller = true;
+
```
Comment (henriquesimoes) [inline utcaApp/src/afc_timing.cpp:76]: This seems a little bit hackish. We only avoid creating it with `p_decoder_controller` counter to later properly identify and treat it on `readEnum`, right?

I feels like defining a parameter as user of decoder controller should be disassociated with such counter. Don't have a concrete proposal, though. Just make it clear with a comment why such hack is interesting for now.

**Review by henriquesimoes — COMMENTED** [review]
Skimming through parameter name changes, they seem consistent.  I should review it thoroughly when it becomes a non-draft.

File: utcaApp/Db/pos_calc.template (hunk 1 above)
Comment (ericonr) [reply utcaApp/Db/pos_calc.template]: Oh, that was entirely accidental. Thanks for noticing it!

File: utcaApp/src/afc_timing.cpp (hunk 2 above)
Comment (ericonr) [reply utcaApp/src/afc_timing.cpp:76]: `p_decoder_controller` is not used as a counter; its main purpose is simply being a variable with a known address.

Maybe I should have another way of specifying that something uses the decodercontroller that doesn't involve the param numbers :thinking: 

I will add a comment for it, though I would note that it's already been done in another module.

**Comment by ericonr** [comment]
Removing draft status because we are interested in deploying the new gateware.

File: utcaApp/Db/afc_timing.template [hunk 3]
```diff
@@ -31,55 +31,86 @@ record(mbbi, "$(P)$(R)RxEnStatus-Mon"){
...
+    field(INP,"@asyn($(PORT))LOCKED_AFC_PHASE")
 }
-record(bi, "$(P)$(R)GT0ClkLocked-Mon"){
+record(bi, "$(P)$(R)AFCFreqLockedLtc-Mon"){
     field(DTYP, "asynInt32")
-    field(DESC, "GT0 Clock Locked Status")
+    field(DESC, "AFC Frequency Locked Latch")
+    field(SCAN,"I/O Intr")
+    field(INP,"@asyn($(PORT))LOCKED_AFC_FREQ_LTC")
+    alias("$(P)$(R)AFCClkLockedLtc-Mon")
```
Comment (guilhermerc) [inline utcaApp/Db/afc_timing.template:61]: Maybe mention that you kept the support for the previous PV interface?

**Review by guilhermerc — APPROVED** [review]
From my bench tests, this seems to be working fine.

File: utcaApp/src/afc_timing.cpp (hunk 2 above)
Comment (henriquesimoes) [reply utcaApp/src/afc_timing.cpp:76]: > its main purpose is simply being a variable with a known address

That's true. I was thinking more about how its content turn out to be filled by `createParam`.

> I will add a comment for it, though I would note that it's already been done in another module.

I indeed didn't notice this was already done. Well, feel free not to do so, and rather let this pending if we turn out changing how to specify this.

**Review by henriquesimoes — APPROVED** [review]
Just took a more careful look on PV changes, and I didn't notice any inconsistency.
